src/models/Order.js

```diff
@@ -115,6 +111,31 @@ OrderSchema.pre('findOne', async function (next) {
   console.log('Pre-findOne finished');
   next();
 });
+
+// send order completion email when the order status changes to "payment_successful"
+OrderSchema.post('findOneAndUpdate', async function (doc) {
+  try {
+    if (this.getUpdate().$set.orderStatus === 'payment_successful') {
+      const user = await User.findById(doc.user);
```

Comment:
We don't use this variable, so this line (`const user = await ...`) can be removed